static/app/views/starfish/modules/databaseModule/index.tsx

```diff
@@ -10,94 +9,118 @@ import {
   PageErrorAlert,
   PageErrorProvider,
 } from 'sentry/utils/performance/contexts/pageError';
-import {MutableSearch} from 'sentry/utils/tokenizeSearch';
+import {useQuery} from 'sentry/utils/queryClient';
+import {useLocation} from 'sentry/utils/useLocation';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import {getMainTable} from 'sentry/views/starfish/modules/databaseModule/queries';
+import {getDateFilters} from 'sentry/views/starfish/utils/dates';
 
 import DatabaseChartView from './databaseChartView';
 import DatabaseTableView, {DataRow} from './databaseTableView';
 import QueryDetail from './panel';
 
-type Props = {
-  location: Location;
-};
+const HOST = 'http://localhost:8080';
 
-type State = {
-  action: string;
-  table: string;
-  transaction: string;
-  selectedRow?: DataRow;
-};
+function DatabaseModule() {
+  const location = useLocation();
+  const [action, setAction] = useState<string>('ALL');
+  const [table, setTable] = useState<string>('ALL');
+  const [rows, setRows] = useState<{next?: DataRow; prev?: DataRow; selected?: DataRow}>({
+    selected: undefined,
+    next: undefined,
+    prev: undefined,
+  });
 
-class DatabaseModule extends Component<Props, State> {
-  state: State = {
-    action: 'ALL',
-    transaction: '',
-    table: 'ALL',
-  };
+  const tableFilter = table !== 'ALL' ? `domain = '${table}'` : undefined;
+  const actionFilter = action !== 'ALL' ? `action = '${action}'` : undefined;
+
+  const pageFilter = usePageFilters();
+  const {startTime, endTime} = getDateFilters(pageFilter);
+  const DATE_FILTERS = `
+  greater(start_timestamp, fromUnixTimestamp(${startTime.unix()})) and
+  less(start_timestamp, fromUnixTimestamp(${endTime.unix()}))
+`;
 
-  handleSearch(query) {
-    const conditions = new MutableSearch(query);
-    const transactionValues = conditions.getFilterValues('transaction');
-    if (transactionValues.length) {
-      return this.setState({transaction: transactionValues[0]});
-    }
-    if (conditions.freeText.length > 0) {
-      // so no need to wrap it here
-      return this.setState({transaction: conditions.freeText.join(' ')});
-    }
-    return this.setState({transaction: ''});
-  }
+  const {
+    isLoading: isTableDataLoading,
+    data: tableData,
+    isRefetching: isTableRefetching,
+  } = useQuery<DataRow[]>({
+    queryKey: ['endpoints', action, table, pageFilter.selection.datetime],
+    queryFn: () =>
+      fetch(
+        `${HOST}/?query=${getMainTable(
+          startTime,
+          DATE_FILTERS,
+          endTime,
+          '',
```

Comment:
Is this the transaction name? This will break https://github.com/getsentry/sentry/pull/48028/

Reply:
Yup that is the transaction name, because search was missing in the current version.
Yeah good call, that would break that PR, but it would be pretty ez to combine them tho, If you merge that PR i can fix up the conflicts.

Reply:
Conflicts should be resolved, search bar is working properly.